### DISCUSSION TITLE
Brew is not identified as a command inside  postinstall script of `pkgbuild` (MacOS)

### DISCUSSION THREAD

**Discussion by hxvxt2** [Homebrew/discussions#20] [Everyday usage]
Hello Guys,

I have a small shell script that installs `Brew` if it is not already installed. 

```bash
#!/bin/bash -v

which -s brew
if [[ $? != 0 ]] ; then
    # Install Homebrew
    echo "Installing Brew..."
    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
    echo "I don't think brew is installed..."

else
    brew update
fi
```
The script runs fine when I run it as a standalone shell script. So I tried to create a package using the `pkgbuild` utility. But once inside the package the same script fails to identify Brew as a command, even though `Brew` is installed. 

I enabled install logs and I can see `Sep 29 22:48:32 hxvxt2 installd[80453]: ./postinstall: /tmp/PKInstallSandbox.42j7tU/Scripts/org.mybinary.0.0.4.sDKM7K/postinstall: line 16: brew: command not found` 

What am I missing? 

Thanks!

**Comment by gromgit** [comment]
I assume the error you got is from your postinstall script trying to run `brew` _after_ your script is run.

It's almost certainly a `PATH` issue. Add `export PATH=/usr/local/bin:$PATH` at the start of your postinstall script to fix it.

**Reply by hxvxt2** [reply]
@gromgit I thought so. So, I included the absolute path to the `brew` i.e `/usr/local/bin/brew` and got ```ERROR: Running Homebrew as root is an extremely dangerous``` error. Got the same error after including `export PATH=/usr/local/bin:$PATH`

**Reply by SMillerDev** [reply]
Yeah, running brew as root is not supported (you'd probably need to sudo back to regular user permissions before running it.)

**Reply by hxvxt2** [reply]
But I'm not running it as root. I mean not intentionally. The installer needs the `sudo` password at the launch, I think my postinstall as a consequence of this is also running as root. But within my script, I'm not elevating user's privilege in any way. The same user is able to run the script as a standalone script. But now within the postinstall script of `.pkg`

I tried `su - myusername` to revoke any privilege elevation. But I still get the same message.

**Reply by SMillerDev** [reply]
I think pkg installs always run as root.

**Comment by gromgit** [comment]
You should be able to preload sudo credentials for the user by running `sudo -v` before running your Homebrew install script.

**Reply by hxvxt2** [reply]
@gromgit So, I made some progress. I think the problem is that the binary I'm trying to install via `Brew` needs `Sudo` password but `Brew` needs to run without `Sudo`. Both of these requirements in the single line. This is what I am doing right now:-

`su - $USER -c "brew cask install mybinary"` 

And this this is the error that I get

```
Sep 30 17:51:29 hxvxt2 installd[90276]: ./postinstall: sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper  
Sep 30 17:51:29 hxvxt2 installd[90276]: ./postinstall: Error: Failure while executing; `/usr/bin/sudo -E -- /usr/bin/env LOGNAME=koba USER=koba USERNAME=koba/usr/sbin/installer -pkg /usr/local/Caskroom/mybinary.pkg -target /` exited with 1. Here's the output:
Sep 30 17:51:29 hxvxt2 installd[90276]: ./postinstall: sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
Sep 30 17:51:29 hxvxt2 installd[90276]: ./postinstall: ==> Purging files for Cask mybinary
```

**Reply by SMillerDev** [reply]
https://github.com/Homebrew/discussions/discussions/2 should also have some info regarding this
